lib/router/transition-state.ts

```diff
@@ -42,40 +42,41 @@ export default class TransitionState<T extends Route> {
     // The prelude RSVP.resolve() async moves us into the promise land.
     return Promise.resolve(null, this.promiseLabel('Start transition'))
       .then(resolveOneRouteInfo, null, this.promiseLabel('Resolve route'))
-      .catch(handleError, this.promiseLabel('Handle error'));
+      .catch(handleError, this.promiseLabel('Handle error'))
+      .then(() => {
+        return currentState;
+      });
 
     function innerShouldContinue() {
-      return Promise.resolve(
-        shouldContinue(),
-        currentState.promiseLabel('Check if should continue')
-      ).catch(function (reason) {
+      if (shouldContinue()) {
+        return true;
+      } else {
         // We distinguish between errors that occurred
         // during resolution (e.g. before"Model/model/afterModel),
         // and aborts due to a rejecting promise from shouldContinue().
         wasAborted = true;
-        return Promise.reject(reason);
-      }, currentState.promiseLabel('Handle abort'));
+        throw new Error('Transition aborted');
```

Comment:
given https://github.com/tildeio/router.js/pull/306/commits/e83c782d88d45c41cdfc5f6d1b23d74e15ad3481#r518934850 is this else unreachable?

Reply:
guess this is ultimately removed, so it doesn't matter.